src/frontends/tensorflow/src/op/where.cpp

```diff
@@ -14,9 +15,10 @@ namespace tensorflow {
 namespace op {
 
 OutputVector translate_where_op(const NodeContext& node) {
-    auto x = node.get_input(0);
-    auto non_zero = make_shared<NonZero>(x);
-    auto transpose_order = make_shared<Constant>(element::i64, Shape{2}, vector<int64_t>{1, 0});
+    default_op_checks(node, 1, {"Where"});
+    auto condition = node.get_input(0);
+    auto non_zero = make_shared<NonZero>(condition, element::i64);
+    auto transpose_order = make_shared<Constant>(element::i32, Shape{2}, vector<int32_t>{1, 0});
```

Comment:
Is it necessary to have transpose order i32 while NonZero output is i64?

Reply:
Yes, the output of Where is always of i64 type. `i32` type tensor is used for transposing order,  a different role. The reason I chose i32 is to spend less memory.